subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/GradleInceptionPerformanceTest.groovy

```diff
@@ -51,4 +59,47 @@ class GradleInceptionPerformanceTest extends AbstractCrossVersionPerformanceTest
         tasks  | _
         'help' | _
     }
+
+    @Category(PerformanceExperiment)
+    @Unroll
+    def "buildSrc change in #testProject comparing gradle"() {
+        given:
+        runner.testProject = testProject
+        runner.tasksToRun = ['help']
+        runner.targetVersions = ["5.0-20181016235834+0000"]
+        runner.runs = runs
+        runner.args = [
+            "-Djava9Home=${System.getProperty('java9Home')}",
+            "-Pgradlebuild.skipBuildSrcChecks=true"
+        ]
+
+        and:
+        def changingClassFilePath = "buildSrc/${buildSrcProjectDir}src/main/groovy/ChangingClass.groovy"
+        runner.addBuildExperimentListener(new BuildExperimentListenerAdapter() {
+            @Override
+            void beforeInvocation(BuildExperimentInvocationInfo invocationInfo) {
+                new File(invocationInfo.projectDir, changingClassFilePath).tap {
+                    parentFile.mkdirs()
+                    text = """
+                        class ChangingClass {
+                            def value = "${invocationInfo.phase} ${invocationInfo.iterationNumber}"
```

Comment:
:x: It would be better to apply a change to the public interface of the type to account for any compilation avoidance like naming the property after the phase and iterationNumber.